subliminal/providers/itasa.py

```diff
@@ -0,0 +1,390 @@
+        # build the param
+        params = {'apikey': self.apikey, 'q': series}
+
+        # make the search
+        logger.info('Searching show ids with %r', params)
+        r = self.session.get(self.server_url + 'shows/search', params=params, timeout=10)
+        r.raise_for_status()
+        root = ParserBeautifulSoup(r.content, ['lxml-xml', 'lxml', 'html.parser']).find('root')
+
+        if int(root.data.count.text) == 0:
+            logger.warning('Show id not found: no suggestion')
+            return None
+
+        # Looking for show in first page
+        for show in root.data.shows:
+            if sanitized_string_equal(show.find('name').text.lower(), series.lower()):
+                show_id = int(show.id.text)
+                logger.debug('Found show id %d', show_id)
+
+                return show_id
+
+        # Not in the first page of result try next (if any)
+        next = root.find('next')
+        while next.text != '':
+
+            r = self.session.get(next.text, timeout=10)
+            r.raise_for_status()
+            root = ParserBeautifulSoup(r.content, ['lxml-xml', 'lxml', 'html.parser']).find('root')
+
+            logger.info('Loading suggestion page %s', root.data.page.text)
+
+            # Looking for show in following pages
+            for show in root.data.shows:
+                if sanitized_string_equal(show.find('name').text.lower(), series.lower()):
+                    show_id = int(show.id)
+                    logger.debug('Found show id %d', show_id)
+
+                    return show_id
+
+            next = root.find('next')
+
+        # No matches found
+        logger.warning('Show id not found: suggestions does not match')
+
+        return None
+
+    def get_show_id(self, series, country_code=None):
+        """Get the best matching show id for `series`.
+
+        First search in the result of :meth:`_get_show_ids` and fallback on a search with :meth:`_search_show_id`
+
+        :param str series: series of the episode.
+        :return: the show id, if found.
+        :rtype: int or None
+
+        """
+        series_sanitized = sanitize_string(series).lower()
+        show_ids = self._get_show_ids()
+        show_id = None
+
+        # attempt with country
+        if not show_id and country_code:
+            logger.debug('Getting show id with country')
+            show_id = show_ids.get('%s %s' % (series_sanitized, country_code.lower()))
+
+        # attempt clean
+        if not show_id:
+            logger.debug('Getting show id')
+            show_id = show_ids.get(series_sanitized)
+
+        # search as last resort
+        if not show_id:
+            logger.warning('Series not found in show ids')
+            show_id = self._search_show_id(series)
+
+        return show_id
+
+    def _download_zip(self, sub_id):
+        # download the subtitle
+        logger.info('Downloading subtitle %r', sub_id)
+
+        # log into main site
+        if not self.login_itasa:
```

Comment:
This is a particular to itasa. the site login is needed to download subtitles (give session id).
Is it better to do both login in initialize?